applications/app/controllers/AtomPageController.scala

```diff
@@ -0,0 +1,63 @@
+package controllers
+
+import com.gu.contentapi.client.model.v1.ItemResponse
+import common._
+import model.Cached.{RevalidatableResult, WithoutRevalidationResult}
+import model._
+import play.api.mvc._
+import play.twirl.api.Html
+import scala.concurrent.Future
+import contentapi.ContentApiClient
+import model.content._
+import views.html.fragments.atoms.{ media => MediaAtomBody, storyquestions => StoryQuestionsAtomBody }
+
+class AtomPageController(contentApiClient: ContentApiClient)(implicit context: ApplicationContext) extends Controller with Logging with ExecutionContexts {
+
+  def render(atomType: String, id: String) = Action.async { implicit request =>
+
+    lookup(s"atom/$atomType/$id") map {
+      case Left(model: MediaAtom) =>
+        (MediaAtomPage(model), MediaAtomBody(model, displayCaption = false, mediaWrapper = Some(MediaWrapper.EmbedPage)))
+      case Left(model: StoryQuestionsAtom) =>
+        (StoryQuestionsAtomPage(model), StoryQuestionsAtomBody(model, isAmp = false))
+      case Left(model: Quiz) => /* noop */
+      case Left(model: InteractiveAtom) => /* noop */
+      case Left(model: RecipeAtom) => /* noop */
+      case Left(model: ReviewAtom) => /* noop */
+      case Right(other) => (other, Nil)
+    } map {
+      case (atom: AtomPage, body: Html) => renderAtom(atom, body)
+      case (error: Result, _) => renderOther(error)
+    }
+  }
+
+  private def lookup(path: String)(implicit request: RequestHeader): Future[Either[Atom, Result]] = {
+    val edition = Edition(request)
+    contentApiClient.getResponse(contentApiClient.item(path, edition)) map make map {
+      case Some(x) => Left(x)
+      case _ => Right(NotFound)
+    } recover convertApiExceptions
+  }
+
+  def make(apiAtom: ItemResponse): Option[Atom] = {
```

Comment:
`make` might not be the best name. When calling it with `AtomPageController.make(response)`, it would be hard to correctly guess that it returns an `Atom`